backends/p4tools/modules/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -293,8 +293,8 @@ void ExprStepper::evalInternalExternMethodCall(const IR::MethodCallExpression *c
          [this](const IR::MethodCallExpression * /*call*/, const IR::Expression * /*receiver*/,
                 IR::ID & /*methodName*/, const IR::Vector<IR::Argument> *args,
                 const ExecutionState &state, SmallStepEvaluator::Result &result) {
-             const auto *blockRef = args->at(0)->expression->checkedTo<IR::PathExpression>();
-             const auto *block = state.findDecl(blockRef);
+             const auto *blockRef = args->at(0)->expression->checkedTo<IR::StringLiteral>();
```

Comment:
Currently, we treat `PathExpressions` and `IR::Members` as references to an existing object in the symbolic map or available declarations. However, in this case we are passing in a string literal that does not exist (the parameter names from the arch specification). This literal is effectively just a prefix to the variables we want to actually copy, which are the individual struct members. If we kept the PathExpressions here, the interpreter would try to resolve them in `resolveMethodCallArguments` and throw an error. 

I think this distinction between StringLiterals (strings we do not resolve) and StateVariables (PathExpressions/Member strings we do resolve) is cleaner because it makes it clear that the input value is a concrete expression (a string prefix) and not a reference. 

I would love to eventually only work with StateVariables and get rid of PathExpressions/Members entirely. Simply to make clear what is meant to be a reference to an object, and what is meant to be a string literal.

Reply:
I see. Although there are still two categories of string literals that could be worth distinguishing (although it might be just too pedantic): the once that refer to existing objects in the program (although not defined in the P4 sense) and arbitrary literals used by P4 programmer.